### PR TITLE
release v0.17.0

### PR DESCRIPTION
**Pull request by aaronjudd**
## New Features

* Accounts Search
* Orders Search
* Order confirmation email template
  - full order items, summary, address
* Search Sorting / Filtering

### Product Detail Page
 - revisions for all elements
 - publish changes workflow
 - private / public visibility
 - archival
 - new React components
 - is still a "simple" product with variants
 - enable or disable revision product type
 - retains existing `product-variants` for non-revision

## Changes:

* Docker build improvements 
* Updated and completed translations (#1495)
  - Arabic
  - Chinese
  - French
  - German
  - Hebrew
  - Romanian contributed by @EpifaniaCollection
  - Russian
  - Spanish

* some languages that need translators are disabled by default
* experimental data migrations (#247)
* experimental acceptance tests + browserstack

## Contributors
@EpifaniaCollection @markdowney @joseconstela @spencern @priezz @callmephilip

### CODE REVIEW

File: imports/plugins/core/ui/client/containers/editContainer.js [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+            hasChange = true;
+          }
+        }
+      }
+    } else if (this.props.data.__draft) {
+      status = "warning";
+
+      tooltip = (
+        <span>
+          <Translation defaultValue="Unpublised Changes" i18nKey="revisions.unpublishedChanges" />
```
Comment (zenweasel) [inline imports/plugins/core/ui/client/containers/editContainer.js]: typo in `defaultValue` - unpublished

File: imports/plugins/core/ui/client/containers/editContainer.js [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+          } else if (typeof this.props.field === "string" && this.props.field === diff.path[0]) {
+            hasChangedField = true;
+          }
+
+          if (hasChangedField) {
+            status = "warning";
+
+            tooltip = (
+              <span>
+                <Translation defaultValue="Unpublised Changes" i18nKey="revisions.unpublishedChanges" />
```
Comment (zenweasel) [inline imports/plugins/core/ui/client/containers/editContainer.js]: Typo in `defaultValue` - Unpublished

File: imports/plugins/core/ui/client/containers/tagListContainer.js [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+          onGetSuggestions={this.handleGetSuggestions}
+          onMoveTag={this.handleMoveTag}
+          onNewTagSave={this.handleNewTagSave}
+          onNewTagUpdate={this.handleNewTagUpdate}
+          onTagRemove={this.handleTagRemove}
+          onTagSave={this.handleTagSave}
+          onTagUpdate={this.handleTagUpdate}
+          suggestions={this.state.suggestions}
+          tags={this.tags}
+          tooltip="Unpublised changes"
```
Comment (zenweasel) [inline imports/plugins/core/ui/client/containers/tagListContainer.js]: Typo: Unpublished

File: imports/plugins/core/ui/client/containers/tagListContainer.js [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+    }
+
+    return false;
+  }
+
+  handleNewTagSave = (tag) => {
+    if (this.productId && this.canSaveTag(tag)) {
+      Meteor.call("products/updateProductTags", this.productId, tag.name, null, (error) => {
+        if (error) {
+          return Alerts.toast("Tag already exists, duplicate add failed.", "error");
```
Comment (zenweasel) [inline imports/plugins/core/ui/client/containers/tagListContainer.js]: Doesn't this need i18n?

File: imports/plugins/core/ui/client/containers/tagListContainer.js [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+        return true;
+      });
+    }
+  }
+
+  handleTagRemove = (tag) => {
+    if (this.productId) {
+      Meteor.call("products/removeProductTag", this.productId, tag._id, (error) => {
+        if (error) {
+          Alerts.toast("Tag already exists, duplicate add failed.", "error");
```
Comment (zenweasel) [inline imports/plugins/core/ui/client/containers/tagListContainer.js]: Why does this alert say the add failed when we are handling removes?

File: imports/plugins/included/product-variant/client/templates/products/productDetail/variants/variant.js [hunk 6]
```diff
@@ -1,6 +1,7 @@
 import { Reaction } from "/client/api";
 import { ReactionProduct } from "/lib/api";
-import { EditButton } from "/imports/plugins/core/ui/client/components";
+import { EditButton, VisibilityButton } from "/imports/plugins/core/ui/client/components";
+import { Meteor } from "meteor/meteor";
```
Comment (zenweasel) [inline imports/plugins/included/product-variant/client/templates/products/productDetail/variants/variant.js]: Import order?

File: imports/plugins/included/product-variant/client/templates/products/productDetail/variants/variantForm/variantForm.js [hunk 7]
```diff
@@ -1,8 +1,10 @@
 import { Reaction, i18next } from "/client/api";
 import { ReactionProduct } from "/lib/api";
+import { applyProductRevision } from "/lib/api/products";
```
Comment (zenweasel) [inline imports/plugins/included/product-variant/client/templates/products/productDetail/variants/variantForm/variantForm.js]: import order?

File: imports/plugins/included/product-variant/client/templates/products/productDetail/variants/variantList/variantList.js [hunk 8]
```diff
@@ -178,6 +178,19 @@ Template.variantList.helpers({
...
+  ChildVariantRevisionButton() {
+    const variant = Template.currentData();
+    const parentVariant = Products.findOne(variant.ancestors[1]);
+
+    return {
+      component: VisibilityButton,
+      toggleOn: variant.isVisible,
+      onClick() {
+        ReactionProduct.toggleVisibility(variant);
+        console.log("would attempt to publish product", variant._id);
```
Comment (zenweasel) [inline imports/plugins/included/product-variant/client/templates/products/productDetail/variants/variantList/variantList.js]: Remove console.log

**Comment by zenweasel** [comment]
👍

**Comment by aaronjudd** [comment]
👍

**Pull request merged**